### ISSUE TITLE
[Editor] Async method within ScriptComponent constructor crashes editor

### ISSUE THREAD

**Issue by Eideren** [xenko3d/xenko#220]
**Release Type**: Official Release/GitHub (please choose appropriate option)

**Version**: 3.0.0.6 commit [ac4e7e0](https://github.com/xenko3d/xenko/commit/ac4e7e03b8e2e0e68005589bd52f7f2a2aba7b1a)

**Platform(s)**: Windows Editor

**Describe the bug**
Occurs when adding it to an entity, tested with an async method running a Task.Delay of one second,the crash handler then pops up.

**To Reproduce**
See description.

**Expected behavior**
Fix the issue or detect and prohibit those kinds of implementation.

**Screenshots**

**Log and callstacks**
[editor.txt](https://github.com/xenko3d/xenko/files/2419545/editor.txt)

**Additional context**

**Comment by Kryptos-FR** [comment]
Unfortunately there is no way to "hide" the constructor in a script.
So the rule is: don't do anything in a constructor. Just don't.
And it's pretty hard to recover from an exception in a constructor. I'm not sure what we can do.

**Comment by jonstelly** [comment]
This is somewhat similar to https://github.com/xenko3d/xenko/issues/209

While testing out my issue I noticed it happens also if you create any field/property initializers that fail when run inside the editor.

Guidance/rules to not put anything in the constructor are completely reasonable, but even documentation of those rules with big flashing warning signs will get missed by some people so are there ways we could catch these errors to give a meaningful message/feedback?

try/catch around the Activator.CreateInstance() call in ObjectFactoryRegistry.NewInstance() to throw a friendly/helpful message, whatever behavior you guys feel is reasonable for the Editor (message box or something else?).  Maybe even changing synchronization contexts during the Activator.CreateInstance() call to ensure that async errors in the constructor would get handled the same way?

I'm happy to work on this if you guys have a certain behavior you think makes sense.

**Comment by Eideren** [comment]
@jonstelly Yep `CreateInstance()`, also initialize fields and auto properties, like a new() would.

I say let's not force users to adopt engine-specific behavior when programming, let them use their C# habits and notify them of issues instead of forcing them out of it. If we're going that way we would have to force them to avoid using field and auto prop initialization, 'doesn't seem like a good idea at all.

On your thread you mentioned MessageBox showing up twice, the editor spawns up to five instances of the object once we add it to an entity, here's a bunch of [stacktrace](https://github.com/xenko3d/xenko/files/2428215/InitStacktrace.txt) demonstrating that, not sure why that is but you can use that file to investigate if you want.
3 of their instantiation comes from `Xenko.Core.Assets.DirtyFlagChangedDelegate` and diverges from there: one of them runs `[...].Assets.Analysis.AssetDependencyManager.Session_AssetDirtyChanged`, another `[...].Assets.Tracking.AssetSourceTracker.Session_AssetDirtyChanged`.

**Comment by Eideren** [comment]
Regarding my last message, those instances are created for each version of the users games that xenko/stride spawns for visualization and serialization purposes.
I should have time to take a look at this and #209 tomorrow.
